R/plots.R

```diff
@@ -207,20 +207,35 @@ resetSearch <- function(pkgNamespaces) {
 getMappingPlottingData <- function(study = study, modelID = modelID, featureID = featureID, testID = testID, libraries = NULL) {
   mapping <- getMapping(study, libraries = libraries)
 
-  # Checking requirements for mapping
-  model_features <- mapping[["defaults"]][modelID[1]] [!is.na(mapping[["defaults"]][modelID[1]])]
+  # at least two mapping names are not named after modelID.
+  if (sum(!names(mapping) %in% modelID) > 1) {
+    stop("Two or more mapping list elements are not named after modelIDs.")
+  }
+
+  # if modelID[1] is not present in names(mapping) and there is a 'default'
+  # mapping, then use the 'default'. Else, use mapping named after modelID[1].
+  if (!modelID[1] %in% names(mapping) && sum(!names(mapping) %in% modelID) == 1) {
+    mapping_name <- names(mapping)[!names(mapping) %in% modelID]
+  } else if (modelID[1] %in% names(mapping)) {
+    mapping_name <- modelID[1]
+  } else {
+    return()
+  }
```

Comment:
For the above code, is there a reason you can't do something like the following?

```R
mapping <- getMapping(study, modelID = modelID, libraries = libraries)
```

In other words, can we offload the logic of testing for valid modelIDs and the presence of a `"default"` option to `getMapping()`?

Reply:
If I got the point correctly, this statement below:

 mapping_name <- names(mapping)[!names(mapping) %in% modelID]
 
 gets whatever mapping name that is not in modelID. Mapping name "default" is recommended only in the documentation of addMapping, but user can call it whatever (s)he wants. Still, I am requesting the mapping list element to be named. Is that the issue, i.e., whether user can't simply provide a mapping with no name?